common/dev/tmp431.c

```diff
@@ -0,0 +1,99 @@
+#include "tmp431.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "sensor.h"
+#include "hal_i2c.h"
+#include "libutil.h"
+
+#define I2C_DATA_SIZE 5
+#define RANGE_0_127 0
+#define RANGE_m55_150 1
+
+static uint8_t temperature_range = 0xFF;
+
+uint8_t tmp431_read(uint8_t sensor_num, int *reading)
```

Comment:
Should we check to see if `sensor_num` is in a valid range?

0 <= sensor_num < MAX_NUM_SENSORS ?

Reply:
We have fixed this and do the forced update.
The sensor_num is unsigned type so only judge if it's less than MAX_NUM_SENSORS. 
Please review it. Thanks.